lib/ClangImporter/ImportType.cpp

```diff
@@ -2660,6 +2658,13 @@ static ParamDecl *getParameterInfo(ClangImporter::Implementation *impl,
   if (param->hasDefaultArg() && !isInOut &&
       !isa<clang::CXXConstructorDecl>(param->getDeclContext()) &&
       impl->isDefaultArgSafeToImport(param)) {
+
+    // If this was a default argument expression with template parameter, return
+    // std::nullopt
+    if (isTemplated) {
+      return std::nullopt;
+    }
```

Comment:
This makes the entire function not available in Swift, but I don't think that's what we really want (and that would be a source compatibility breakage). I think we should just drop the default expression for this parameter by adding `&& !isTemplated` to the if-condition above.

Reply:
Done